calculator/interfaces/CalculatorDateAndTime.java

```diff
@@ -0,0 +1,125 @@
+package calculator.interfaces;
+import java.time.ZonedDateTime;
+import java.time.ZoneId;
+import java.util.regex.*;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAdjusters;
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.LocalDate;
+public class CalculatorDateAndTime implements Calculator {
+
+	private final Pattern pattern = Pattern.compile("^([\\+\\-])\\s*((\\d*)y)?\\s*((\\d*)m)?\\s*((\\d*)d)?\\s*((\\d*)h)?\\s*((\\d*)min)?\\s*((\\d*)s)?$");
+	
+	private ZonedDateTime zone = ZonedDateTime.now();
+
+	public CalculatorDateAndTime (Zone zone) {
+		this.zone = ZonedDateTime.now(ZoneId.of(zone.idZone));
+	}
+	public String toString () {
+		return zone.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
+	}
+    public String whenSunday () {
+        return "Sunday is " + zone.with(TemporalAdjusters.next(DayOfWeek.SUNDAY)).format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+    }
+    public void showSecondsSince (String input) {
+        Pattern yearMonthDay = Pattern.compile("^showSecondsSince(\\d{1,4})\\s?([1-9]|1[0-2])?\\s?([1-9]|1[0-9]|2[0-9]|3[01])?$");
+        Matcher matcher = yearMonthDay.matcher(input);
+        Duration secondsSince;
+        matcher.matches();
+
+        if (matcher.group(2) != null && matcher.group(3) != null && matcher.group(1) != null) {
+            LocalDate localDate = LocalDate.of(Integer.valueOf(matcher.group(1)).intValue(), Integer.valueOf(matcher.group(2)).intValue(), Integer.valueOf(matcher.group(3)).intValue());
+            secondsSince =  Duration.between(zone, zone.minusYears(localDate.getYear()).minusMonths(localDate.getMonthValue()).minusDays(localDate.getDayOfMonth()));
+            System.out.println("Seconds: " + secondsSince.getSeconds());
+        } 
+        else if (matcher.group(1) != null && matcher.group(2) != null) {
+            YearMonth yearMonth = YearMonth.of(Integer.valueOf(matcher.group(1)).intValue(), Integer.valueOf(matcher.group(2)).intValue());
+            secondsSince = Duration.between(zone, zone.minusYears(yearMonth.getYear()).minusMonths(yearMonth.getMonthValue()));
+            System.out.println("Seconds: " + secondsSince.getSeconds());
+        }       
+        else if (matcher.group(1) != null) {
+            Year year = Year.of(Integer.valueOf(matcher.group(1)).intValue());
+            secondsSince = Duration.between(zone, zone.minusYears(zone.getYear() - year.getValue()));
+            System.out.println("Seconds: " + secondsSince.getSeconds());
+        }
+        else {
+            System.out.println("Bad date");
+        }
+    }
+    public void printHelp() {
```

Comment:
decide if you put space between function name and parameters
printHelp() or printHelp ()
calculate (String number) or calculate(String number)?
make every method follow same format